src/marqo/_httprequests.py

```diff
@@ -52,64 +63,106 @@ def send_request(
         with warnings.catch_warnings():
             if not self.config.cluster_is_remote:
                 warnings.simplefilter('ignore', InsecureRequestWarning)
-            try:
-                request_path = self.config.url + '/' + path
-                if isinstance(body, (bytes, str)):
-                    response = http_method(
-                        request_path,
-                        timeout=self.config.timeout,
-                        headers=req_headers,
-                        data=body,
-                        verify=to_verify
-                    )
-                else:
-                    response = http_method(
-                        request_path,
-                        timeout=self.config.timeout,
-                        headers=req_headers,
-                        data=json.dumps(body) if body else None,
-                        verify=to_verify
-                    )
-                return self.__validate(response)
-            except requests.exceptions.Timeout as err:
-                raise BackendTimeoutError(str(err)) from err
-            except requests.exceptions.ConnectionError as err:
-                raise BackendCommunicationError(str(err)) from err
+            
+            for attempt in range(max_retry_attempts + 1):
+                try:
+                    request_path = self.config.url + '/' + path
+                    if isinstance(body, (bytes, str)):
+                        response = http_method(
+                            request_path,
+                            timeout=self.config.timeout,
+                            headers=req_headers,
+                            data=body,
+                            verify=to_verify
+                        )
+                    else:
+                        response = http_method(
+                            request_path,
+                            timeout=self.config.timeout,
+                            headers=req_headers,
+                            data=json.dumps(body) if body else None,
+                            verify=to_verify
+                        )
+                    return self.__validate(response)
+                except requests.exceptions.Timeout as err:
+                    raise BackendTimeoutError(str(err)) from err
+                except requests.exceptions.ConnectionError as err:
+                    if (attempt == max_retry_attempts):
+                        raise BackendCommunicationError(str(err)) from err
+                    else:
+                        logger.info(f"BackendCommunicationError encountered... Retrying request to {request_path}.")
```

Comment:
We could have more detail like `Retry attempt 2 of 3...`

Reply:
Yep I had that discussion with @pandu-k as well and it didn't really matter too much for the initial implementation and most retry mechanisms are executed silently in terms of a user's perspective.

If that is what is preferred, I can implement that as well.